Sources/TDS/Message/TDSMessage+Tokens.swift

```diff
@@ -174,7 +223,178 @@ extension TDSMessages {
+                guard let len = messageBuffer.readInteger(endianness: .little, as: LongLen.self) else {
+                    throw TDSError.protocolError("Error while reading length")
+                }
+                length = Int(len)
+            case .char, .varchar, .nchar, .nvarchar, .binary, .varbinary:
+                guard let len = messageBuffer.readInteger(endianness: .little, as: UShortCharBinLen.self) else {
+                    throw TDSError.protocolError("Error while reading length")
+                }
+                length = Int(len)
+            case .date:
+                length = 3
+            case .tinyInt, .bit:
+                length = 1
+            case .smallInt:
+                length = 2
+            case .intType, .smallDateTime, .real, .smallMoney:
+                length = 4
+            case .money, .datetime, .float, .bigInt:
+                length = 8
+            case .nullType:
+                length = 0
+            default:
+                guard let len = messageBuffer.readInteger(endianness: .little, as: ByteLen.self) else {
+                    throw TDSError.protocolError("Error while reading length.")
+                }
+                length = Int(len)
+            }
+
+            var collationData: [UInt8] = []
+            if (dataType.isCollationType()) {
+                guard let collationBytes = messageBuffer.readBytes(length: 5) else {
+                    throw TDSError.protocolError("Error while reading COLLATION.")
+                }
+                collationData = collationBytes
+            }
+
+            var precision: Int?
+            if (dataType.isPrecisionType()) {
+                guard
+                    let p = messageBuffer.readInteger(as: UInt8.self),
+                    p <= 38
+                else {
+                    throw TDSError.protocolError("Error while reading PRECISION.")
+                }
+                precision = Int(p)
+            }
+
+            var scale: Int?
+            if (dataType.isScaleType()) {
+                guard let s = messageBuffer.readInteger(as: UInt8.self) else {
+                    throw TDSError.protocolError("Error while reading SCALE.")
+                }
+
+                if let p = precision {
+                    guard s <= p else {
+                        throw TDSError.protocolError("Invalid SCALE value. Must be less than or equal to precision value.")
+                    }
+                }
+
+                scale = Int(s)
+            }
+
+            // TODO: Read [TableName] and [CryptoMetaData]
+            var tableName: String?
+            switch dataType {
+            case .text, .nText, .image:
+                var parts: [String] = []
+                guard let numParts = messageBuffer.readInteger(as: UInt8.self) else {
+                    throw TDSError.protocolError("Error while reading NUMPARTS.")
+                }
+
+                for i in 0...numParts - 1 {
```

Comment:
this could be `for _ in 0...numParts - 1` since `i` never used.

Reply:
done.